app/src/main/java/com/example/sharingang/ItemsListFragment.kt

```diff
@@ -20,32 +21,48 @@ class ItemsListFragment : Fragment() {
 
     private lateinit var binding: FragmentItemsListBinding
     private val viewModel: ItemsViewModel by activityViewModels()
+    private val uiViewModel: UIViewModel by activityViewModels()
+
     @Inject
     lateinit var currentUserProvider: CurrentUserProvider
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
+        savedInstanceState: Bundle?,
     ): View {
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_items_list, container, false)
         binding.viewModel = viewModel
 
         val adapter = viewModel.setupItemAdapter(currentUserProvider.getCurrentUserId())
         binding.itemList.adapter = adapter
+        viewModel.addObserver(viewLifecycleOwner, adapter, ItemsViewModel.OBSERVABLES.ORDERED_ITEMS)
         binding.itemList.layoutManager = GridLayoutManager(context, 2)
-        viewModel.addObserver(viewLifecycleOwner, adapter, ItemsViewModel.OBSERVABLES.ALL_ITEMS)
 
         viewModel.setupItemNavigation(viewLifecycleOwner, this.findNavController(),
-                {item -> ItemsListFragmentDirections.actionItemsListFragmentToDetailedItemFragment(item)})
+            { item -> ItemsListFragmentDirections.actionItemsListFragmentToDetailedItemFragment(item) })
 
         binding.swiperefresh.setOnRefreshListener { viewModel.refresh() }
         viewModel.refreshing.observe(viewLifecycleOwner, {
             if (!it) {
                 binding.swiperefresh.isRefreshing = false
+                orderItems() // in the case an item is added, need to add redo the ordering
             }
         })
+        binding.orderCategorySpinner.setSelection(uiViewModel.orderByPosition.value!!)
+        binding.orderAscendingDescending.setSelection(uiViewModel.ascendingDescendingPosition.value!!)
```

Comment:
Are the `!!` necessary here? Both of the livedata don't contain null values, so this seems odd

Reply:
Yeah I was confused too, Android Studio keeps telling me it needs that